lib/krikri/harvesters/marc_xml_harvester.rb

```diff
@@ -0,0 +1,106 @@
+module Krikri::Harvesters
+
+  class MarcXMLHarvester
+    include Krikri::Harvester
+
+    attr_reader :opts
+
+    def initialize(opts = {})
+      @max_records = opts.fetch(:max_records, :unlimited)
```

Comment:
I'll implement that and send a PR, then maybe these can be rebased on the generalized solution.